src/shim/runtime/v2/rune/v2/service.go

```diff
@@ -0,0 +1,1036 @@
+		return nil, err
+	}
+
+	switch containerType {
+	case shimtypes.PodSandbox:
+		if sandboxNamspace != shimtypes.KubeSystem {
+			agentID, err := readAgentIdFile(path)
+			if err != nil {
+				return nil, err
+			}
+
+			err = cleanupAgentContainer(ctx, agentID)
+			if err != nil {
+				return nil, err
+			}
+		}
+	case shimtypes.PodContainer:
+		if sandboxNamspace != shimtypes.KubeSystem {
+			os.RemoveAll(filepath.Join(ContainerBase, s.id))
+		}
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"root":    root,
+		"path":    path,
+		"ns":      ns,
+		"runtime": runtime,
+	}).Debug("Container Cleanup()")
+
+	r := process.NewRunc(root, path, ns, runtime, "", false)
+	if err := r.Delete(ctx, s.id, &runcC.DeleteOpts{
+		Force: true,
+	}); err != nil {
+		logrus.WithError(err).Warn("failed to remove runc container")
+	}
+	if err := mount.UnmountAll(filepath.Join(path, "rootfs"), 0); err != nil {
+		logrus.WithError(err).Warn("failed to cleanup rootfs mount")
+	}
+
+	pid, err := runcC.ReadPidFile(filepath.Join(path, process.InitPidFile))
+	if err != nil {
+		logrus.WithError(err).Warn("failed to read init pid file")
+	}
+
+	return &taskAPI.DeleteResponse{
+		ExitedAt:   time.Now(),
+		ExitStatus: 128 + uint32(unix.SIGKILL),
+		Pid:        uint32(pid),
+	}, nil
+}
+
+func setOCIRuntime(ctx context.Context, r *taskAPI.CreateTaskRequest) (err error) {
+	var opts options.Options
+	// read options to get OCI Runtime
+	if r.Options != nil {
+		v, err := typeurl.UnmarshalAny(r.Options)
+		if err != nil {
+			return err
+		}
+		opts = *v.(*options.Options)
+	}
+	if opts.BinaryName != RuntimeClass {
+		opts.BinaryName = RuntimeClass
+		r.Options, err = typeurl.MarshalAny(&opts)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Create a new initial process and container with the underlying OCI runtime
+func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *taskAPI.CreateTaskResponse, err error) {
+	logrus.WithField("container", r.ID).Debug("Create() start")
+	defer logrus.WithField("container", r.ID).Debug("Create() end")
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	err = setOCIRuntime(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+	setLogLevel(logLevel)
+
+	container, err := create(ctx, s, r)
+	if err != nil {
+		return nil, err
+	}
+
+	s.containers[r.ID] = container
+
+	s.send(&eventstypes.TaskCreate{
+		ContainerID: r.ID,
+		Bundle:      r.Bundle,
+		Rootfs:      r.Rootfs,
+		IO: &eventstypes.TaskIO{
+			Stdin:    r.Stdin,
+			Stdout:   r.Stdout,
+			Stderr:   r.Stderr,
+			Terminal: r.Terminal,
+		},
+		Checkpoint: r.Checkpoint,
+		Pid:        uint32(container.Pid()),
+	})
+
+	return &taskAPI.CreateTaskResponse{
+		Pid: uint32(container.Pid()),
+	}, nil
+}
+
+// Start a process
+func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (*taskAPI.StartResponse, error) {
+	logrus.WithField("container", r.ID).Debug("Start() start")
+	defer logrus.WithField("container", r.ID).Debug("Start() end")
```

Comment:
All these paired log calls make me wonder if adding tracing might be better medium term?